### PR TITLE
Gaya's Bowling Challenge in JS

### PR DESCRIPTION
**Pull request by gdrmvrm**
The program returns the perfect score. However, I was unsure how to tests mocks of the 'Add Bonus Points' function in the 'Calculate Bonus Points' function (in the Score Card Class) as the 'Calculate Bonus Points' function it does not return values.  I would much appreciate if I can get feedback on how this is done. Also, any feedback on my JS code setup and how I can improve will be very helpful! Thanks!

### CODE REVIEW

File: lib/frame.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    return this.tenthFrame;
+  }
+
+  setTenthFrame() {
+    this.tenthFrame = true;
+  }
+
+  addRoll(roll) {
+    if (roll > 10) {
+      throw "Maximum score of 10 allowed";
```
Comment (adamwoodcock98) [inline lib/frame.js:32]: Nice job catching this edge case!

**Review by adamwoodcock98 — COMMENTED** [review]
Really nice job Gaya!

You used some really nice features of JS that we've not covered yet like the try-catch blocks, so well done.

Lots of tests which is fantastic!

Didn't have enough time to clone the project and run it, but it seems very good and you've covered the tenth frame and the perfect score so great job!

**Review by leoht — COMMENTED** [review]
Really good work! Re: mocking, you've correctly mocked the `Frame` class in the tests, and I left a few comments on things that I see could be improved — for example, you could simplify / extract the bonus calculation further (see my comments on the code below). Let me know if you have any more questions on it :)
